### PR TITLE
MM-22265  - Restrict the WS "update_team" event broadcast

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Fixed sending of update_team to every user

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-22265

### CODE REVIEW

File: app/team.go [hunk 1]
```diff
@@ -253,6 +253,7 @@ func (a *App) sendTeamEvent(team *model.Team, event string) {
 	sanitizedTeam.Sanitize()
 
 	message := model.NewWebSocketEvent(event, "", "", "", nil)
+	message.Broadcast.ContainsSensitiveData = true
```
Comment (jupenur) [inline app/team.go]: This prevents the event from being sent to anyone other than system admins. Is that reasonable? How does the webapp behave for non-admins when a team is edited?

Comment (reflog) [reply app/team.go]: @jupenur crap. I think you are right. Might have to rethink this

Comment (ethervoid) [reply app/team.go]: Good point. Also, this affects UPDATE_TEAM, DELETE_TEAM and RESTORE_TEAM events

Comment (reflog) [reply app/team.go]: @jupenur done. Please review

File: app/team.go [hunk 2]
```diff
@@ -252,7 +252,12 @@ func (a *App) sendTeamEvent(team *model.Team, event string) {
 	*sanitizedTeam = *team
 	sanitizedTeam.Sanitize()
 
-	message := model.NewWebSocketEvent(event, "", "", "", nil)
+	teamId := "" // no filtering by teamId by default
+	if event == model.WEBSOCKET_EVENT_UPDATE_TEAM {
```
Comment (jupenur) [inline app/team.go]: Currently the webapp `/select_team` view for joining and creating teams seems to update itself from websockets. How does this change affect it? Should this check account for public teams visible to all users on the server?

**Pull request merged**